flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/expressions/resolver/rules/ResolveCallByArgumentsRule.java

```diff
@@ -108,24 +116,34 @@ private static class ResolvingCallVisitor
         public List<ResolvedExpression> visit(UnresolvedCallExpression unresolvedCall) {
             final FunctionDefinition definition;
             // clean functions that were not registered in a catalog
-            if (!unresolvedCall.getFunctionIdentifier().isPresent()) {
+            if (unresolvedCall.getFunctionIdentifier().isEmpty()) {
                 definition =
                         prepareInlineUserDefinedFunction(unresolvedCall.getFunctionDefinition());
             } else {
                 definition = unresolvedCall.getFunctionDefinition();
             }
 
-            final String name =
+            final String functionName =
                     unresolvedCall
                             .getFunctionIdentifier()
                             .map(FunctionIdentifier::toString)
                             .orElseGet(definition::toString);
 
             final Optional<TypeInference> typeInference = getOptionalTypeInference(definition);
 
+            // Reorder named arguments and add replacements for optional ones
+            if (definition == BuiltInFunctionDefinitions.ASSIGNMENT) {
+                throw new ValidationException(
```

Comment:
I am probably missing something. It indicates  `if the definition is an assignment error`, afterwards` call executeAssignment. ` Does this mean the definition is not an assignment when we call executeAssignment.
If this is intended maybe a comment to explain - as it looks strange.

Reply:
This line is meant for cases where ASSIGNMENT is used outside of a function call, i.e. as a top level expression. I will add more meaningful comment here.